demo/api/src/db/migrations/Migration20230131155247.ts

```diff
@@ -0,0 +1,13 @@
+import { Migration } from "@mikro-orm/migrations";
+
+export class Migration20230131155247 extends Migration {
+    async up(): Promise<void> {
+        this.addSql('drop index "unique_name_in_root_folder_index"');
+        this.addSql('drop index "unique_name_in_folder_index"');
+
+        this.addSql('create unique index "unique_name_in_root_folder_index" on "DamFile" ("name", "scope_domain") where "folderId" IS NULL');
```

Comment:
This indices need to be changed based on whether we use a scope or not. Having to add this migration sucks. Should we remove the constraints and add a validation in code instead?

Reply:
Validation in code we have to do anyway. should be enough imho, especailly as those filenames are basically just "decoration".

Reply:
> Also, we decided that the system should handle duplicate filenames automatically (by adding `-copy${i}` to the end). This is already the case for uploads but not (yet) when moving files.

What should happen when the `filesService.updateById` is called with a name that already exists in the folder? Should we throw an error?

Reply:
I've decided to add a validation in `FilesService.updateByEntity`, which is used by all update operations: cee6c2571a9842692e55a9f12424d46fdcb152b6

We may add a filename collision handling in the future on top of that. Nevertheless do we need this validation anyway.